src/angular-business/table/table/table-data-source.ts

```diff
@@ -190,33 +220,45 @@ export class SbbTableDataSource<T> extends DataSource<T> {
-   * filter matching.
+   * at least one occurrence in that string.
+   *
+   * # TableFilter variant
+   * Checks if a data object matches the data source's filter object. If several columns are defined,
+   * the and operator is applied. The '_' property of the TableFilter can be used to search globally in all columns
+   * (like the string variant above).
+   *
+   * By default, the filter string has its whitespace trimmed and the match is case-insensitive.
+   * May be overridden for a custom implementation of filter matching.
    * @param data Data object used to check against the filter.
-   * @param filter Filter string that has been set on the data source.
+   * @param filter Filter string or Object which extends TableFilter that has been set on the data source.
    * @returns Whether the filter matches against the data
    */
-  filterPredicate: (data: T, filter: string) => boolean = (data: T, filter: string): boolean => {
-    // Transform the data into a lowercase string of all property values.
-    const dataStr = Object.keys(data)
-      .reduce((currentTerm: string, key: string) => {
-        // Use an obscure Unicode character to delimit the words in the concatenated string.
-        // This avoids matches where the values of two columns combined will match the user's query
-        // (e.g. `Flute` and `Stop` will match `Test`). The character is intended to be something
-        // that has a very low chance of being typed in by somebody in a text field. This one in
-        // particular is "White up-pointing triangle with dot" from
-        // https://en.wikipedia.org/wiki/List_of_Unicode_characters
-        return currentTerm + (data as { [key: string]: any })[key] + '◬';
-      }, '')
-      .toLowerCase();
-
-    // Transform the filter by converting it to lowercase and removing whitespace.
-    const transformedFilter = filter.trim().toLowerCase();
-
-    return dataStr.indexOf(transformedFilter) !== -1;
+  filterPredicate: (data: T, filter: TFilter) => boolean = (data: T, filter: TFilter): boolean => {
+    const tableData = data as { [key: string]: any };
+    const tableFilter: { [key: string]: any } & TableFilter =
+      typeof filter === 'string'
+        ? ({ _: filter } as TableFilter)
+        : (filter as { [key: string]: any } & TableFilter);
+
+    let matchesGlobal = true;
+    if (typeof tableFilter._ === 'string') {
+      matchesGlobal = matchesStringCaseInsensitive(reduceObjectToString(tableData), tableFilter._!);
+    }
+
+    const matchesEveryColumnSearched = Object.keys(tableFilter)
+      .filter(key => key !== '_')
+      .filter(
+        key => typeof tableFilter[key] !== 'undefined' && ('' + tableFilter[key]).trim() !== ''
+      ) // filter filter columns which have an empty search string
+      .filter(key => typeof tableData[key] !== 'undefined' && tableData[key] !== null) // filter table columns which are not defined
```

Comment:
Maybe combine the three filters into one filter?
This is a bit nitpicky; From a performance perspective, this is called often and for multiple data entries. For each of these, the filter object is iterated up to three times.
What do you think?